src/CalcManager/Ratpack/exp.cpp

```diff
@@ -342,11 +342,32 @@ void powratNumeratorDenominator(PRAT *px, PRAT y, uint32_t radix, int32_t precis
         DUPRAT(originalResult, pxPow);
         powratcomp(&originalResult, oneoveryDenom, radix, precision);
                 
+        // ##################################
+        // Find number of digits after decimal point
+        // and prepare multiplier for rounding
+        // ##################################
+        PRAT resultMultiplier = nullptr;
+        PRAT resultDigits = nullptr;
+        PNUMBER numResult = RatToNumber(pxPow, radix, precision);
+        if (-numResult->exp < precision)
+        {
+            DUPRAT(resultDigits, longtorat(-numResult->exp / 2));
```

Comment:
`longtorat` returns a `PRAT`, which should be captured by a variable and then destroyed later to avoid a memory leak.